packages/vite/src/node/server/hmr.ts

```diff
@@ -71,6 +71,52 @@ interface PropagationBoundary {
 }
 
 export interface HotChannelClient {
+  send(payload: HotPayload): void
+}
+/** @deprecated use `HotChannelClient` instead */
+export type HMRBroadcasterClient = HotChannelClient
+
+export type HotChannelListener<T extends string = string> = (
+  data: InferCustomEventPayload<T>,
+  client: HotChannelClient,
+  invoke: 'send' | `send:${string}` | undefined,
+) => void
```

Comment:
I got a type error with this combination.

```ts
declare let listener: HotChannelListener
declare let payload: CustomPayload;

listener(payload.data, { send: () => {} }, payload.invoke);
// Argument of type '"send" | `send:${string}` | "response" | `response:${string}` | undefined' is not assignable to parameter of type '"send" | `send:${string}` | undefined'.
//  Type '"response"' is not assignable to type '"send" | `send:${string}` | undefined'.ts(2345)
```

It doesn't look like users need to be aware of the complicated `invoke` typing. Is this only for internal type safety? I think it's not that bad even if it's just `invoke: string | undefined`.

Reply:
I changed to `CustomPayload['invoke']` so should be fine now. I guess `CustomPayload['invoke']` will give users more information on what should be passed.